client/app/components/QueryLink.jsx

```diff
@@ -21,10 +21,12 @@ function QueryLink({ query, visualization, readOnly }) {
     return query.getUrl(false, hash);
   };
 
+  const QueryLinkWrapper = props => (readOnly ? <span {...props} /> : <Link href={getUrl()} {...props} />);
```

Comment:
It feels like the styles could diverge a little / the spacing might get off (just because one's a styled component and one's a plain ol' `span`. If that doesn't happen, then this is fine 👍 .

Reply:
It doesn't in this case and _shouldn't_, but I agree it can become a problem. Any ideas on how to prevent problems? This is, however, the proper approach.

Reply:
I think as long as `Link` remains an unstyled component, there's no problem.